services/spar/test-integration/Test/Spar/Scim/UserSpec.hs

```diff
@@ -660,6 +664,93 @@ testScimCreateVsUserRef = do
       (call . delete $ brig . paths ["i", "users", toByteString' uid])
         !!! const 202 === statusCode
 
+testCreateUserTimeout :: TestSpar ()
+testCreateUserTimeout = do
+  env <- ask
+
+  email <- randomEmail
+  scimUser <- randomScimUser <&> \u -> u {Scim.User.externalId = Just $ fromEmail email}
+  (_owner, tid) <- call $ createUserWithTeam (env ^. teBrig) (env ^. teGalley)
+  tok <- registerScimToken tid Nothing
+
+  testFlow tid tok scimUser email
+  -- creating the same user again should behave the same
+  testFlow tid tok scimUser email
+  where
+    testFlow tid tok scimUser email = do
+      env <- ask
+      let brig = env ^. teBrig
+
+      scimStoredUser <- createUser tok scimUser
+      let handle = Handle . Scim.User.userName . Scim.value . Scim.thing $ scimStoredUser
+
+      inv <- call $ getInvitation brig email
+
+      -- Wait for user and invitation to expire
+      let setTeamInvitationTimeout = 5 -- Must match the value in brig.integration.yaml
+      threadDelay $ (setTeamInvitationTimeout + 1) * 1_000_000
+
+      -- User doesn't show in scim
+      do
+        listUsers tok (Just (filterBy "userName" $ fromHandle handle)) >>= \users ->
+          liftIO $ users `shouldBe` []
+        listUsers tok (Just (filterBy "externalId" $ fromEmail email)) >>= \users ->
+          liftIO $ users `shouldBe` []
+
+      -- User should fail to register invitation
+      do
+        call $ do
+          void $ do
+            Just inviteeCode <- getInvitationCode brig tid (inInvitation inv)
+            post
+              ( brig . path "/register"
+                  . contentJson
+                  . json (acceptWithName (Name "Bob") email inviteeCode)
+              )
+              <!! const 201 =/= statusCode
+
+      -- User still doesn't show in scim
```

Comment:
I would put this block in a definition under a nice name, and call that definition twice, to reduce code duplication.

It's a question of taste, though.